core/src/test/java/org/opensearch/sql/analysis/ExpressionAnalyzerTest.java

```diff
@@ -228,6 +231,42 @@ public void qualified_name_with_qualifier() {
     analysisContext.pop();
   }
 
+  @Test
+  public void qualified_name_with_metadata_field_success() {
+    analysisContext.push();
+    analysisContext.peek().define(new Symbol(Namespace.INDEX_NAME, "index_alias"), STRUCT);
+
+    assertAnalyzeEqual(DSL.ref("_id", STRING), qualifiedNameWithMetadata("index_alias", "_id"));
```

Comment:
To be honest, I don't like how assertAll doesn't show you the specific error of the test, and you need to drill down into the code stack to find what is failing.  I'd prefer to keep this explicit since each test corresponds to a different scenario and different line of code.